ipfs-gateway-finder/src/main.rs

```diff
@@ -554,8 +569,11 @@ async fn probe_gateway(
     num_tries: u32,
     timeout: Duration,
 ) -> Result<()> {
-    let mut url = Url::parse(gateway_url.replace(":hash", &cid).as_str())?;
+    let formatted_url = format!("{}/ipfs/{}", gateway_url, cid);
```

Comment:
Oh, I see, they removed the placeholders in the official list.
For backwards compatibility, could you check if the placeholder is there and only append this if it's not? I have a bunch of code relying on the old format...